priv/script.sh

```diff
@@ -2,441 +2,489 @@
-                fish_add_path /opt/homebrew/bin
-            else
-                (
-                    echo
-                    echo 'eval "$(/opt/homebrew/bin/brew shellenv)"'
-                ) >>$config_file
-            fi
-            eval "$(/opt/homebrew/bin/brew shellenv)"
-        else
-            if [ "$current_shell" = "fish" ]; then
-                fish_add_path /usr/local/bin
-            else
-                (
-                    echo
-                    echo 'eval "$(/usr/local/bin/brew shellenv)"'
-                ) >>$config_file
-            fi
-            eval "$(/usr/local/bin/brew shellenv)"
-        fi
-        ;;
-    "mise")
-        curl https://mise.run | sh
-
-        case $current_shell in
-        "bash")
-            echo 'eval "$(~/.local/bin/mise activate bash)"' >>$config_file
-            ;;
-        "fish")
-            echo '~/.local/bin/mise activate fish | source' >>$config_file
-            ;;
-        "zsh")
-            echo 'eval "$(~/.local/bin/mise activate zsh)"' >>$config_file
-            ;;
-        esac
-
-        export PATH="$HOME/.local/bin:$PATH"
-        ;;
-    "Phoenix")
-        mise exec -- mix local.hex --force
-        mise exec -- mix local.rebar --force
-        mise exec -- mix archive.install --force hex phx_new $phoenix_version
-        ;;
-    "PostgreSQL")
-        if [ "$os_type" = "macOS" ]; then
-            brew install gcc readline zlib curl ossp-uuid
-        else
-            case "$package_manager" in
-            "apt")
-                sudo apt-get update
-                sudo apt-get install -y linux-headers-generic build-essential libssl-dev libreadline-dev zlib1g-dev libcurl4-openssl-dev uuid-dev icu-devtools
-                ;;
-            "dnf")
-                sudo dnf groupinstall -y "Development Tools"
-                sudo dnf install -y kernel-headers openssl-devel readline-devel zlib-devel libcurl-devel libuuid-devel libicu-devel
-                ;;
-            "pacman")
-                sudo pacman -Sy --noconfirm linux-headers base-devel openssl readline zlib curl util-linux icu
-                ;;
-            "apk")
-                sudo apk add --no-cache linux-headers build-base openssl-dev readline-dev zlib-dev curl-dev util-linux-dev icu-dev
-                ;;
-            esac
-        fi
-        mise use -g -y postgres@$postgres_version
-        ;;
-    "Xcode Command Line Tools")
-        xcode-select --install
-        ;;
+    Homebrew | Erlang)
+      printf "Installing %s... (this might take a while)\n" "$1"
+      ;;
     *)
-        printf "Invalid name argument on install: %s\n" "$1"
-        exit 1
-        ;;
+      printf "Installing %s...\n" "$1"
+      ;;
     esac
+
+    install "$1"
+  fi
 }
 
-maybe_install() {
-    if already_installed "$1"; then
-        printf "%s is already installed. Skipping...\n" "$1"
-    else
-        printf "Installing %s...\n" "$1"
+install_shell_scripts() {
```

Comment:
This is the real change here. Instead of modifying *just* `~/.bashrc`, `~/.zshrc`, or `~/.config/fish/config.fish`, we're creating `~/.config/phx_tools/phx_tools.{sh,fish}` with the appropriate configuration changes.

The `sh` version is written to work with both Bash and zsh.